src/Kiota.Builder/Writers/Php/PhpConventionService.cs

```diff
@@ -61,7 +61,7 @@ public override string TranslateType(CodeType type)
             {
                 "boolean" => "bool",
                 "double" or "decimal" => "float",
-                "integer" or "int32" or "int64" => "int",
```

Comment:
There is a custom Byte class in the abstractions which should be used to handle `byte`. For the other unsigned integers, I think we should have proper validation since just `int`  is too broad especially for unsigned values.

Reply:
yep it seems the logic here "broadens" or "shrinks" the types:
- Double is float64
- decimal float128
- int64 should be int64
- missing byte (uint8)
- missing int8

But then again maybe the php type system is more flexible.

Reply:
> yep it seems the logic here "broadens" or "shrinks" the types:
> 
> * Double is float64
> * decimal float128
> * int64 should be int64
> * missing byte (uint8)
> * missing int8
> 
> But then again maybe the php type system is more flexible.

PHP's way more flexible. The type system represents numbers as `int` or `float`. Number of bits depends on whether the platform is 32-bit (int and float would be 32-bit) or 64-bit

Not sure how we'd best represent decimal values since those would be 128-bit

Reply:
the CPU instruction width is one thing. The type precision is another. But in that context how would somebody running on 32 bits CPUs represent an int64? My concern is that if an API returns int64 to some client running 32 bits, they'll lose information in the process.

Reply:
Did some reading on primitive types...
- PHP only supports unsigned integers so `byte` types will be mapped to string. Developers can use PHP extension libraries: [GMP](https://www.php.net/manual/en/intro.gmp.php) or [BCMath](https://www.php.net/manual/en/intro.bc.php) to work with unsigned ints
- Same extensions can be used for float128/decimal types hence the mapping to string

Reply:
thanks for the precision. What happens to somebody in PHP manipulating int64s on a x86(32 bits) architecture?

Reply:
PHP converts it to a `float`. Thinking we should have an additional Kiota config option for processor architecture?

Reply:
> There is a custom Byte class in the abstractions which should be used to handle `byte`. For the other unsigned integers, I think we should have proper validation since just `int` is too broad especially for unsigned values.

Also on extending the native type system with custom wrapper types...
I understand that it could potentially prevent a developer from setting an invalid value but I find it unsustainable to try to implement a custom type system in a case like this.
Noticed TypeScript doesn't take the approach of custom types. Thoughts @baywet @SilasKenneth ?

Reply:
> > There is a custom Byte class in the abstractions which should be used to handle `byte`. For the other unsigned integers, I think we should have proper validation since just `int` is too broad especially for unsigned values.
> 
> Also on extending the native type system with custom wrapper types... I understand that it could potentially prevent a developer from setting an invalid value but I find it unsustainable to try to implement a custom type system in a case like this. Noticed TypeScript doesn't take the approach of custom types. Thoughts @baywet @SilasKenneth ?

I understand why a custom Date and Time class because they wrap around a native DateTime and serialize the native DateTime to different appropriate string values.
But in cases where we don't serialize/deserialize differently, seems unsustainable

Reply:
> Noticed TypeScript doesn't take the approach of custom types.

That's because everything is a number in typescript/javascript, there's no specialized types in the runtime. Now if you're telling me for example that in PHP all ints are ints, regardless of their size, I'm happy with having just int methods. I simply don't want to end up in a case where somebody running on a 32 bits architecture and receiving a 64 bits length int from the API gets an exception or losses information.

> I understand why a custom Date and Time

So SDK users don't have to provide a time value when using a date only and vice versa. Also so the serialization code "knows" what's the expected precision and we don't get "time shifts" by setting default values.

Reply:
> I simply don't want to end up in a case where somebody running on a 32 bits architecture and receiving a 64 bits length int from the API gets an exception or losses information.



@baywet Added a [platform check](https://github.com/microsoft/kiota/pull/1473/files#diff-b7fdb9f1e9da2811bd67921b68d27d2a75a5b60efde98db9afe697caa04ef5a5R66) so that int64's are represented as `float` on 32-bit architectures

Reply:
I don't think this approach is valid as a 64 bit machine could be used to generate code that'll run on a 32 bits one

Reply:
@Ndiritu sorry, I might have miss expressed myself. I do think we're headed into the wrong direction for that specific problem.
In 2022 I can't imagine shipping an SDK for each CPU architecture. Yes runtimes have to do that, but that's the reason why we leverage runtimes, and the SDKs we're generating are "just" sending HTTP requests, not doing some kind of low level hardware communication.

If PHP cannot reliably carry int64 values across CPU architectures (from a quick research PHP7 and up might in fact [support an int64 type](https://wiki.php.net/rfc/64bit-integer-type)), then we should elect a type from a library that does and wrap it in our own type in abstractions (like some of the Date types in other languages).

I hope this clarifies the intent.

Reply:
@baywet PHP7 doesn't support int64s on 32-bit arch unfortunately. The linked RFC is "Work in Progress".

There are also no packages/extensions I see to handle this.

Some thoughts/options:
- PHP-8 supports union types so int64 can be mapped to `int|float`. This could be added justification for #1526 
- Or we create a custom wrapper type and add to abstractions.

My issue with the two approaches is that it makes an edge case issue make your 64-bit arch types unnecessarily complex for a dev. I feel that for 64-bit archs, int64s should just directly map to int.

Since PHP's type system depends on hardware architecture, I'd argue Kiota should care about it for PHP's sake?

Or it's also worth determining whether we should care about supporting 32-bit architectures? Not sure if they're still used widely or how to find out. Any suggestions?

Reply:
I'd be ok with making the assumptions of "all ints at 64 bits" and "using a kiota SDK requires running on 64 hardware/process" as long as we have conclusive data on the matter:
- what architectures does PHP support outside of x86/32 and 64 bits? how do those other architectures impact ints?
- what's the current distribution of PHP apps by host process architecture?

Adding a 32 bit or target architecture switch to kiota is a non-starter for me because it'd mean multiple SDKs for each architecture, which is a huge maintenance burden.

Reply:
@baywet couldn't find any useful stats on PHP usage by processor architecture. PHP only supports x86 and x64.

From general 32-bit vs 64-bit research - seems the general trend is towards 64-bit processors even in the IoT world.

From PHP stats, ~90% of apps are using a version >= 7.3 which introduced strong typing so 

 So I'd agree that 32-bit support is an edge case

Happy to revert the processor architecture checks.

Reply:
Can you have a conversation with @roinochieng first to make sure the consequences are understood and agreed upon first please?